src/iexamine.cpp

```diff
@@ -2233,6 +2233,26 @@ void iexamine::tree_blackjack(player *p, map *m, const tripoint &examp)
     m->ter_set(examp, t_tree);
 }
 
+void iexamine::tree_birch(player *p, map *m, const tripoint &examp)
+{
+    if(!query_yn(_("Pick %s?"), m->tername(examp).c_str())) {
+        none(p, m, examp);
+        return;
+    }
+    m->spawn_item( p->pos(), "birchbark", rng( 1, 2 ) );
+    m->ter_set(examp, t_tree_birch_harvested);
+}
+
+void iexamine::tree_willow(player *p, map *m, const tripoint &examp)
+{
+    if(!query_yn(_("Pick %s?"), m->tername(examp).c_str())) {
+        none(p, m, examp);
+        return;
+    }
+    m->spawn_item( p->pos(), "willowbark", rng( 1, 2 ) );
+    m->ter_set(examp, t_tree_willow_harvested);
```

Comment:
You can write this as `m->ter_set(examp, "t_tree_willow_harvested");` (note the quotation marks), to avoid having to define `t_tree_willow_harvested` in mapdata.h (this is why the build currently fails). Similar in the other iexamine function.